resources/assets/js/components/FindSkillsModal.tsx

```diff
@@ -0,0 +1,481 @@
+                    className="no-list-style-type"
+                  >
+                    {skillsResults.map((skill) => {
+                      const { id, name, description } = skill;
+                      const isAdded = newSkills.find(
+                        (newSkill) => newSkill.id === skill.id,
+                      );
+                      const isOldSkill =
+                        portal === "applicant" &&
+                        oldSkills.find(
+                          (oldSkill) => oldSkill.id === skill.id,
+                        ) !== undefined;
+                      return (
+                        <li
+                          key={id}
+                          data-h2-grid="b(middle, contained, padded, 0)"
+                        >
+                          <Accordion data-h2-grid-item="b(3of4)">
+                            <Accordion.Btn>
+                              <p
+                                data-h2-font-weight="b(700)"
+                                data-h2-font-style="b(underline)"
+                              >
+                                {name[locale]}
+                                {isAdded && (
+                                  <i
+                                    data-h2-padding="b(left, .5)"
+                                    data-h2-font-color="b(theme-1)"
+                                    aria-hidden="true"
+                                    className="fas fa-check"
+                                  />
+                                )}
+                              </p>
+                            </Accordion.Btn>
+                            <Accordion.Content>
+                              <p data-h2-focus>{description[locale]}</p>
+                            </Accordion.Content>
+                          </Accordion>
+                          {isOldSkill ? (
+                            <button
+                              data-h2-button=""
+                              data-h2-grid-item="b(1of4)"
+                              disabled
+                              type="button"
+                            >
+                              <span data-h2-button-label>
+                                {intl.formatMessage(
+                                  messages.disabledSkillButton,
+                                )}
+                              </span>
+                            </button>
+                          ) : (
+                            <button
+                              data-h2-button=""
+                              data-h2-grid-item="b(1of4)"
+                              type="button"
+                              disabled={isOldSkill}
+                              onClick={() => {
+                                // If the skill has been selected then remove it.
+                                // Else, if the has not been selected then add it to addedSkills list.
+                                if (isAdded) {
```

Comment:
There's very similar logic when `setExpandedAccordions` is used, I wonder if a little helper function would be nice, that removes an element from an array its already present, and removes it if it's not.